### ISSUE TITLE
📎 Implement `noAwaitInLoop` - `eslint/no-await-in-loop`

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#3934]
### Description

[no-await-in-loop](https://eslint.org/docs/latest/rules/no-await-in-loop/).

**Want to contribute?** Lets you know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

**Comment by kotasudhakar** [comment]
I want to contribute but I'm new to rust, could you please guide me on this?

**Comment by Conaclos** [comment]
@kotasudhakar
Sure!
First read the [Analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md). It will teach you how to create a new rule.

You can use the `syntax` tab of the [playground](https://biomejs.dev/playground/?code=YQBzAHkAbgBjACAAZgB1AG4AYwB0AGkAbwBuACAAZgAoACkAIAB7AAoAIAAgAGYAbwByACgAOwA7ACkAIAB7AAoAIAAgACAAIAAvAC8AIABTAGgAbwB1AGwAZAAgAGIAZQAgAHIAZQBlAHAAbwByAHQAZQBkAAoAIAAgACAAIABhAHcAYQBpAHQAIABnACgAKQA7AAoAIAAgAH0ACgB9AAoACgBmAHUAbgBjAHQAaQBvAG4AIABoACgAKQAgAHsACgAgACAAZgBvAHIAKAA7ADsAKQAgAHsACgAgACAAIAAgAGEAcwB5AG4AYwAgAGYAdQBuAGMAdABpAG8AbgAgAGkAKAApACAAewAKACAAIAAgACAAIAAgAC8ALwAgAFMAaABvAHUAbABkACAAbgBvAHQAIABiAGUAIAByAGUAZQBwAG8AcgB0AGUAZAAKACAAIAAgACAAIAAgAGEAdwBhAGkAdAAgAGoAKAApADsACgAgACAAIAAgAH0ACgAgACAAfQAKAH0A) to understand the structure of a code. You can also take a look at the `js.ungrammar` file
to have the syntax tree model we use.

The rule will have to [query all nodes](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#query-multiple-nodes) that correspond to loops (`JsDoWhileStatement | JsForInStatement | JsForOfStatement | JsForStatement | JsWhileStatement | JsWithStatement`).

You can take some inspiration of the implementation of `noUnusedPrivateClassMembers`.
Basically starting from the loop node, we will have to traverse all descendant nodes. We will skip a subtree if it is a loop (this avoids double reporting of an `await` expression) or if we encounter a `AnyJsControlFlowRoot` (this will skip function declared in a loop, and others).

When we met a `JsAwaitExpression` we can directly return its range as rule's state.
This allows us to precisely report where the `await` is.

Let me know if you need more help.
